test/language/statements/class/subclass/class-definition-null-proto-this.js

```diff
@@ -3,42 +3,37 @@
 /*---
 esid: sec-runtime-semantics-classdefinitionevaluation
 description: >
-  The `this` value of a null-extending class is automatically initialized
+  The `this` value of a null-extending class isn't automatically initialized
 info: |
-  The behavior under test was introduced in the "ES2017" revision of the
-  specification and conflicts with prior editions.
-
   Runtime Semantics: ClassDefinitionEvaluation
 
   [...]
-  5. If ClassHeritageopt is not present, then
-     [...]
-  6. Else,
-     [...]
-     b. Let superclass be the result of evaluating ClassHeritage.
-  [...]
-  15. If ClassHeritageopt is present and superclass is not null, then set F's
-      [[ConstructorKind]] internal slot to "derived".
+  15. If ClassHeritageopt is present, then set F's [[ConstructorKind]] internal slot to "derived".
   [...]
 
-  9.2.2 [[Construct]]
+  12.2.2.1 Runtime Semantics: Evaluation
+  PrimaryExpression : this
+  1. Return ? ResolveThisBinding( ).
 
+  8.3.4 ResolveThisBinding ( )
+  [...]
+  2. Return ? envRec.GetThisBinding().
+  
+  8.1.1.3.4 GetThisBinding ( )
   [...]
-  5. If kind is "base", then
-     a. Let thisArgument be ? OrdinaryCreateFromConstructor(newTarget,
-        "%ObjectPrototype%").
+  3. If envRec.[[ThisBindingStatus]] is "uninitialized", throw a ReferenceError exception.
   [...]
 ---*/
 
-var thisVal, instance;
-
 class C extends null {
   constructor() {
-    thisVal = this;
+    // Use an arrow function to access the `this` binding of the class constructor.
+    assert.throws(ReferenceError, () => {
+      this;
+    });
   }
 }
 
-instance = new C();
-
-assert.sameValue(instance instanceof C, true);
-assert.sameValue(instance, thisVal);
+assert.throws(ReferenceError, function() {
+  new C();
+});
```

Comment:
again a suggestion for a follow up PR: this is not what we are really asserting in this test, so a try catch block would avoid ambiguity.